full/src/test/java/apoc/custom/CypherProceduresClusterTest.java

```diff
@@ -14,29 +13,27 @@
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
-import static org.junit.Assume.assumeTrue;
 
 public class CypherProceduresClusterTest {
 
     private static TestcontainersCausalCluster cluster;
 
     @BeforeClass
     public static void setupCluster() {
-        TestUtil.ignoreException(() ->  cluster = TestContainerUtil
-                .createEnterpriseCluster(3, 1, Collections.emptyMap(), MapUtil.stringMap("apoc.custom.procedures.refresh", "100")),
-                Exception.class);
-        Assume.assumeNotNull(cluster);
-        assumeTrue("Neo4j Cluster should be up-and-running", cluster.isRunning());
```

Comment:
I'm worried about this type of lines. Is spanning the docker container synchronous? Cause normally it isn't, and that would mean we need to bring back this line with retries. Something in the lines of:

```
eventually(cluster.isRunning(), backoff = 10s, maxTime = 1min)
```

Reply:
Thanks for pointing this out.

There was a way of doing this already in the codebase for Neo4jContainerExtension. If you look at this [constructor method](https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/4.4/test-utils/src/main/java/apoc/util/Neo4jContainerExtension.java#L44) then you'll see that it has a WaitStrategy which will only return once some conditions are met, or it will timeout and throw an error after 2 minutes.

The TestContainerCausalCluster setup we have is instantiating multiple Neo4jContainerExtension under the hood and will thus wait for them to all come up.

I had however forgotten to add a wait Strategy for my new GoogleCloudStorageContainerExtension and MySQLContainerExtension

Reply:
Since I made the comment above a lot has changed for the Neo4jContainerExtension because we introduced the  .withWaitForNeo4jDatabaseReady function (and I'm removing the old logic from the constructor method in this PR).